source/common/event/libevent_scheduler.h

```diff
@@ -40,8 +41,22 @@ class LibeventScheduler : public Scheduler {
    */
   event_base& base() { return *libevent_; }
 
+  /**
+   * Start writing stats once thread-local storage is ready to receive them (see
+   * ThreadLocalStoreImpl::initializeThreading).
+   */
+  void initializeStats(DispatcherStats* stats_);
+
 private:
+  static void onPrepare(evwatch*, const evwatch_prepare_cb_info* info, void* arg);
+  static void onCheck(evwatch*, const evwatch_check_cb_info*, void* arg);
+
   Libevent::BasePtr libevent_;
+  DispatcherStats* stats_; // stats owned by the containing DispatcherImpl
```

Comment:
Nit: I have a general preference for `{}` initializing these if not set in the constructor, even if not necessary. Too much time wasted chasing down uninitialized bugs :)